scib/metrics/isolated_labels.py

```diff
@@ -227,7 +227,7 @@ def max_f1(adata, label_key, cluster_key, label, argmax=False):
         # AWS score between isolated label vs rest
         adata.obs[iso_label_key] = adata.obs[label_key] == isolated_label
         adata.obs["silhouette_temp"] = silhouette_samples(
-            adata.obsm[embed], adata.obs[iso_label_key]
+            adata.obsm[embed], adata.obs[label_key]
```

Comment:
Since the ASW is run on the complete data and would always return the same silhouette scores, it might make sense to compute the scores once for the complete dataset and then aggregate it per isolated label score. That would require passing those scores to `score_isolated_label` (or adding them e.g. to `adata.uns` so they can be accessed within `score_isolated_label`), but I'd argue it's worth it, particularly for larger datasets.

Reply:
should work by just checking if the column exists, no?

Reply:
this could however be tricked by just adding a column silhouette_temp=1

Reply:
true, should be per obs. But I think it would be better to explicitly pass it to the function, just to be sure